src/util/environment.rs

```diff
@@ -0,0 +1,141 @@
+use std::env;
```

Comment:
I wanted to separate it into a different module to hide the impl details and this seemed abstract enough to put under util module. Can I put it in the build module but under inline module to achieve the same visibility semantics (so the build/cargo modules can't access private fields)?